src/mmw/apps/geoprocessing_api/urls.py

```diff
@@ -14,7 +14,7 @@
 urlpatterns = [
     url(r'^token/', views.get_auth_token,
         name="authtoken"),
-    url(r'analyze/land/$', views.start_analyze_land,
+    url(r'analyze/land/(?P<nlcd_year>\w*)/?$', views.start_analyze_land,
```

Comment:
I see in commit bb107ae that we documented this breaking change to the API. I do not know much about how API consumers we have. Is is worth considering setting up a 302 from `analyze/land/` to `analyze/land/2011_2011/`? Alternatively, is there a design reason that is making us choose a path parameter over a backwards compatible optional query string parameter (for example `analyze/land/?nlcd_year=2019_2016`)?

Reply:
I went with the path parameter to ensure it is explicitly specified every time, and because there's precedent for this with the DRB 2100 endpoint:

https://github.com/WikiWatershed/model-my-watershed/blob/ae94ac6020b08d6163e43f99cfde35b9b413346c/src/mmw/apps/geoprocessing_api/urls.py#L36-L38

While falling back to 2011 would preserve backwards compatibility, as of this work we'll be switching to 2019 as the "default" dataset for all future work. We _could_ have the default be 2019, and then while no API users will have their systems break, but their values will be different. This ensures that any existing users will get a breaking notification, causing them to update their endpoints mindfully.

Looking at API usage:

```sql
SELECT num_api_calls, username, first_name, last_name, email, organization, user_type, country, postal_code
FROM auth_user u
    INNER JOIN user_userprofile p ON u.id = p.user_id
    INNER JOIN (
        SELECT user_id, COUNT(id) AS num_api_calls
        FROM core_requestlog
        WHERE user_id IS NOT NULL
          AND user_id <> 7305 -- MMW Web App User
          AND api = 't'
          AND requested_at > (CURRENT_DATE - INTERVAL '12 months')
        GROUP BY user_id
    ) x ON u.id = x.user_id
ORDER BY num_api_calls DESC;
```

I see only one API user in the last year, and they just explored for a week in June (not sharing further output of the query for privacy reasons). Given the low API usage and lack of clarity on what would be a good default, I decided to introduce this breaking change without a fallback.